internal/model/yaml_test.go

```diff
@@ -0,0 +1,67 @@
+package model
+
+import (
+	"testing"
+
+	"github.com/sahilm/fuzzy"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestYAML_rxFilter(t *testing.T) {
+	uu := map[string]struct {
+		q     string
+		lines []string
+		e     fuzzy.Matches
+	}{
+		"empty-lines": {
+			q:     "foo",
+			lines: []string{},
+			e:     fuzzy.Matches{},
+		},
+		"no-match": {
+			q:     "foo",
+			lines: []string{"bar"},
+			e:     fuzzy.Matches{},
+		},
+		"single-match": {
+			q:     "foo",
+			lines: []string{"foo", "bar", "baz"},
+			e: fuzzy.Matches{
+				{
+					Str:            "foo",
+					Index:          0,
+					MatchedIndexes: []int{0, 3},
+				},
+			},
+		},
+		"multiple-matches": {
```

Comment:
Might be a good idea to add a use cases where the match is contained multiple time in the line and not necessarily space delimited ie "thefoobar fobarbarfofool barfoofo"...